tests/Microsoft.Identity.Test.Unit/PublicApiTests/MtlsPopTests.cs

```diff
@@ -0,0 +1,522 @@
+                       .WithMtlsProofOfPossession() // Enables MTLS PoP
+                       .ExecuteAsync())
+                    .ConfigureAwait(false);
+
+                Assert.AreEqual(MsalError.MtlsPopWithoutRegion, ex.ErrorCode);
+            }
+        }
+
+        [TestMethod]
+        public async Task MtlsPopWithoutAuthorityAsync()
+        {
+            IConfidentialClientApplication app = ConfidentialClientApplicationBuilder
+                            .Create(TestConstants.ClientId)
+                            .WithCertificate(s_testCertificate)
+                            .WithExperimentalFeatures()
+                            .Build();
+
+            // Set WithMtlsProofOfPossession on the request without specifying an authority
+            MsalClientException ex = await AssertException.TaskThrowsAsync<MsalClientException>(() =>
+                app.AcquireTokenForClient(TestConstants.s_scope)
+                   .WithMtlsProofOfPossession() // Enables MTLS PoP
+                   .ExecuteAsync())
+                .ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public void Constructor_ValidCertificate()
+        {
+            var scheme = new MtlsPopAuthenticationOperation(s_testCertificate);
+
+            // Compute the expected KeyId using SHA-256 on the public key
+            var expectedKeyId = ComputeExpectedKeyId(s_testCertificate);
+
+            Assert.AreEqual(expectedKeyId, scheme.KeyId);
+            Assert.AreEqual(Constants.MtlsPoPTokenType, scheme.AccessTokenType);
+        }
+
+        private static string ComputeExpectedKeyId(X509Certificate2 certificate)
+        {
+            // Get the raw public key bytes
+            var publicKey = certificate.GetPublicKey();
+
+            // Compute the SHA-256 hash of the public key
+            using (var sha256 = SHA256.Create())
+            {
+                byte[] hash = sha256.ComputeHash(publicKey);
+                return Base64UrlHelpers.Encode(hash);
+            }
+        }
+
+        [TestMethod]
+        public void GetTokenRequestParams_ExpectedValues()
+        {
+            var scheme = new MtlsPopAuthenticationOperation(s_testCertificate);
+            System.Collections.Generic.IReadOnlyDictionary<string, string> parameters = scheme.GetTokenRequestParams();
+
+            Assert.AreEqual(Constants.MtlsPoPTokenType, parameters[OAuth2Parameter.TokenType]);
+        }
+
+        [TestMethod]
+        public void FormatResult_SetsMtlsCertificate()
+        {
+            var scheme = new MtlsPopAuthenticationOperation(s_testCertificate);
+            var authenticationResult = new AuthenticationResult();
+
+            scheme.FormatResult(authenticationResult);
+
+            Assert.AreEqual(s_testCertificate, authenticationResult.MtlsCertificate);
+        }
+
+        [DataTestMethod]
+        [DataRow("https://login.microsoftonline.com", "Public Cloud")]
```

Comment:
Do you have a test with a non-standard cloud? Like "https://login.myLocalAAD.com/guid" ?

Reply:
added one now @bgavrilMS please check